src/gc/unix/gcenv.unix.cpp

```diff
@@ -7,6 +7,9 @@
 #include <cassert>
 #include <memory>
 #include <pthread.h>
+#ifdef __FreeBSD__
```

Comment:
I can see that we differ in the way to get the thread id here and in the https://github.com/dotnet/coreclr/blob/master/src/pal/src/include/pal/thread.hpp#L838-L844. Could you please unify it? If both ways return the same values, then I'd use the pthread_getthreadid_np at both places. If not, then please let's use the `thr_self` as it is verified to provide the same identifier as LLDB, which we need in the LLDB plugin.